### DISCUSSION TITLE
Isolate table styling from Quarto doc defaults

### DISCUSSION THREAD

**Discussion by AlbertRapp** [rstudio/gt#1113] [Q&A]
# Context

This is a crosspost from [Stackoverflow](https://stackoverflow.com/questions/74349523/avoid-quarto-table-style-in-table-that-contains-images/74359959#74359959). Currently, there is one suggested answer but it relies solely on restyling everything with CSS. That makes using any styling options like `tab_style()` or `tab_options()` somewhat useless. So, I'd like to avoid needing to do everything with CSS.

Is there a way to isolate the table from Quarto's default? I've also found an idea in #1085  by using custom CSS. But once again this would have to restyle everything with CSS by hand. (Using `all: inherit;` in the CSS-file did not work)

# Original Question

Is there a way to create a table inside a Quarto document without applying the doc's theme to the HTML table.

Usually I use `as_raw_html()` to avoid the Quarto style. But in my current use-case, my table contains a custom ggplot which is removed when I use `as_raw_html()`. 

I've tried using [custom output class](https://stackoverflow.com/questions/74026514/how-to-apply-css-style-to-quarto-output) but this didn't take any effect on the table output.

## Reprex

```
library(tidyverse)
library(gt)

filtered_penguins <- palmerpenguins::penguins |>
    filter(!is.na(sex))

penguin_weights <- palmerpenguins::penguins |>
  filter(!is.na(sex)) |>
  group_by(species) |>
  summarise(
    Min = min(body_mass_g),
    Mean = mean(body_mass_g) |> round(digits = 2),
    Max = max(body_mass_g)
  ) |>
  mutate(species = as.character(species), Distribution = species) |>
  rename(Species = species)


plot_density_species <- function(species, variable) {
  full_range <- filtered_penguins |>
    pull({{variable}}) |>
    range()

  filtered_penguins |>
    filter(species == !!species) |>
    ggplot(aes(x = {{variable}}, y = species)) +
    geom_violin(fill = 'dodgerblue4') +
    theme_minimal() +
    scale_y_discrete(breaks = NULL) +
    scale_x_continuous(breaks = NULL) +
    labs(x = element_blank(), y = element_blank()) +
    coord_cartesian(xlim = full_range)
}

penguin_weights |>
  gt() |>
  tab_spanner(
    label = 'Penguin\'s Weight',
    columns = -Species
  ) |>
  text_transform(
    locations = cells_body(columns = 'Distribution'),
    # Create a function that takes the a column as input
    # and gives a list of ggplots as output
    fn = function(column) {
      map(column, ~plot_density_species(., body_mass_g)) |>
        ggplot_image(height = px(50), aspect_ratio = 3)
    }
  ) 
```

## Desired Output

This is the output I get outside of a Quarto document. But inside a Quarto doc, the doc's theme is applied to the table. Using `as_raw_html()` removes the plots.

[![enter image description here][1]][1]

  [1]: https://i.stack.imgur.com/zIkmf.png

**Comment by rich-iannone** [comment]
@AlbertRapp with https://github.com/rstudio/gt/pull/1114 now merged into the dev version of gt, the images in the plots should no longer be removed. I just tested on my local machine and it looks good now:

<img width="1083" alt="quarto_penguins" src="https://user-images.githubusercontent.com/5612024/201212996-e9eeb118-f69b-4b6b-bdbe-84d418b6bf8e.png">

You'll need to make sure you get the juicyjuice package installed (it is new in CRAN, and a new dependency for gt as of this PR; https://cran.r-project.org/web/packages/juicyjuice/index.html). Let me know if this solves the problem for you on your system. The gt package will be updated soon on CRAN as well.

**Reply by AlbertRapp** [reply]
Hi @rich-iannone,

Thank you for this change. It's a real life-saver. I spend some time building my own function that targets all gt CSS classes from a table's code and replaces it by some other name (so that Quarto can't target it). It works but using `as_raw_html()` is easier :)

In any case, I ran some more examples and documented everything in a Quarto book (so that Quarto effects can be seen) at https://guileless-mousse-1c0b8b.netlify.app/

It appears as if sometimes the code chunk that contains `as_raw_html()` still needs to be wrapped in a div with `style="all:initial;"` to prevent some other Quarto options taking effect. Also, some table lines are not formatted correctly with `as_raw_html()`.

I hope these examples are okay even though they're not minimal. I just wanted to try out many customizations to see what will work in Quarto doc.

**Reply by rich-iannone** [reply]
Albert it’s great to hear that the recent changes constitute an improvement. I’d love to hear about the aspects of the table that aren’t immune to the Quarto CSS. I really like the idea of using that particular style rule to further enforce the isolation of CSS. Could you please open another gt Issue with these findings?

**Reply by srowelluf** [reply]
in case its helpful, I did make note of some cases were quarto was messing with gt formatting in the quarto discussion: https://github.com/quarto-dev/quarto-cli/discussions/3357

In quarto 1.2.269 (and 1.1.189)  the headers of a gt table are changed, and there are some font differences; though using as_raw_html(inline_css = TRUE) fixes.
